it/uk/gov/hmrc/nrs/retrieval/NrsRetrievalIntegrationSpec.scala

```diff
@@ -0,0 +1,217 @@
+    }
+  }
+
+  "GET /nrs-retrieval/submission-bundles/:vaultId/:archiveId" should {
+    "succeed" in {
+      wsClient.url(s"$serviceRoot/submission-bundles/vaultId/archiveId").get().futureValue.status shouldBe OK
+    }
+  }
+
+  "HEAD /nrs-retrieval/submission-bundles/:vaultId/:archiveId" should {
+    val url = s"$serviceRoot/submission-bundles/vaultId/archiveId"
+    val requestWithoutHeaders = wsClient.url(url)
+    val requestWithHeader = wsClientWithXApiKeyHeader(url)
+
+    def givenSubmissionBundlesReturns(status: Int, withHeaders: Boolean): Unit =
+      wireMockServer.addStubMapping(
+        mappingBuilder(
+          head(urlPathMatching("/retrieval/submission-bundles/vaultId/archiveId")), withHeaders
+        ).willReturn(aResponse().withStatus(status)).build())
+
+    Seq(
+      (requestWithHeader, true, "the X-API-Key header"),
+      (requestWithoutHeaders, false, "no headers")
+    ).foreach{ case (request, withHeaders, headersLabel) =>
+      "pass through the X-API-Key header if set and return OK" when {
+        s"the request has $headersLabel and the submission bundles request returns OK" in {
+          givenSubmissionBundlesReturns(OK, withHeaders)
+          request.head().futureValue.status shouldBe OK
+        }
+
+       s"the request has $headersLabel and the submission bundles request returns a 2xx status" in {
+          givenSubmissionBundlesReturns(CREATED, withHeaders)
+          request.head().futureValue.status shouldBe OK
+        }
+      }
+
+      "pass through the X-API-Key header if set and return ACCEPTED" when {
+        s"the request has $headersLabel and the submission bundles request returns ACCEPTED" in {
+          givenSubmissionBundlesReturns(ACCEPTED, withHeaders)
+          request.head().futureValue.status shouldBe ACCEPTED
+        }
+      }
+
+      "pass through the X-API-Key header if set and return NOT_FOUND" when {
+        s"the request has $headersLabel and the submit request returns NOT_FOUND" in {
+          givenSubmissionBundlesReturns(NOT_FOUND, withHeaders)
+          request.head().futureValue.status shouldBe NOT_FOUND
+        }
+      }
+
+      "pass through the X-API-Key header if set and return INTERNAL_SERVER_ERROR" when {
+        s"the request has $headersLabel and the submit request returns a 3xx status" in {
+          givenSubmissionBundlesReturns(SEE_OTHER, withHeaders)
+          request.head().futureValue.status shouldBe INTERNAL_SERVER_ERROR
+        }
+
+        Seq(BAD_REQUEST, UNAUTHORIZED, FORBIDDEN).foreach{ status =>
+          s"the request has $headersLabel and the submit request returns a 4xx status $status" in {
+            givenSubmissionBundlesReturns(status, withHeaders)
+            request.head().futureValue.status shouldBe INTERNAL_SERVER_ERROR
+          }
+        }
+      }
+
+      "pass through the X-API-Key header if set and return BAD_GATEWAY" when {
```

Comment:
it's confusing, to say expected return is BAD_GATEWAY and use INTERNAL_SERVER_ERROR internally

Reply:
Indeed, but is the behaviour we had already...

Reply:
which is part of the opinionated behaviour of `http-client` and why I use `wsClient` in the tests.